src/Neo/UInt160.cs

```diff
@@ -123,44 +124,59 @@ public override string ToString()
         /// <summary>
         /// Parses an <see cref="UInt160"/> from the specified <see cref="string"/>.
         /// </summary>
-        /// <param name="s">An <see cref="UInt160"/> represented by a <see cref="string"/>.</param>
+        /// <param name="str">An <see cref="UInt160"/> represented by a <see cref="string"/>.</param>
         /// <param name="result">The parsed <see cref="UInt160"/>.</param>
         /// <returns><see langword="true"/> if an <see cref="UInt160"/> is successfully parsed; otherwise, <see langword="false"/>.</returns>
-        public static bool TryParse(string s, out UInt160 result)
+        public static bool TryParse(string str, out UInt160 result)
         {
-            if (s == null)
+            result = null;
+
+            if (string.IsNullOrWhiteSpace(str)) return false;
+
+            if (str.StartsWith("0x", StringComparison.InvariantCulture))
+                str = str[2..];
+
+            if (str.Length != Length * 2) return false;
+
+            try
             {
-                result = null;
-                return false;
+                var data = new byte[Length];
+                for (var i = 0; i < Length; i++)
+                {
+                    if (!byte.TryParse(str.AsSpan(i * 2, 2), NumberStyles.HexNumber, null, out data[Length - i - 1]))
+                        return false;
+                }
+                result = new(data);
+                return true;
             }
-            if (s.StartsWith("0x", StringComparison.InvariantCultureIgnoreCase))
-                s = s[2..];
-            if (s.Length != Length * 2)
+            catch
             {
-                result = null;
                 return false;
             }
-            byte[] data = new byte[Length];
-            for (int i = 0; i < Length; i++)
-                if (!byte.TryParse(s.Substring(i * 2, 2), NumberStyles.AllowHexSpecifier, null, out data[Length - i - 1]))
-                {
-                    result = null;
-                    return false;
-                }
-            result = new UInt160(data);
-            return true;
+        }
+
+        public static implicit operator UInt160(string s)
+        {
+            return Parse(s);
+        }
+
+        public static implicit operator UInt160(byte[] b)
+        {
+            return new UInt160(b);
         }
 
         public static bool operator ==(UInt160 left, UInt160 right)
         {
-            if (ReferenceEquals(left, right)) return true;
-            if (left is null || right is null) return false;
+            if (left is null || right is null)
+                return Equals(left, right);
             return left.Equals(right);
         }
 
         public static bool operator !=(UInt160 left, UInt160 right)
         {
-            return !(left == right);
+            if (left is null || right is null)
+                return !Equals(left, right);
+            return !left.Equals(right);
```

Comment:
If we use ` [MethodImpl(MethodImplOptions.AggressiveInlining)]` we can preserve the previous code